vm/src/stdlib/sys.rs

```diff
@@ -316,6 +317,57 @@ mod sys {
         vm.write_exception(&mut crate::py_io::PyWriter(stderr, vm), &exc)
     }
 
+    #[pyfunction(name = "__breakpointhook__")]
+    #[pyfunction]
+    pub fn breakpointhook(args: FuncArgs, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        let envar =
+            std::env::var("PYTHONBREAKPOINT").unwrap_or_else(|_| "pdb.set_trace".to_owned());
+
+        if envar.eq("0") {
+            return Ok(vm.ctx.none());
+        };
+
+        let print_unimportable_module_warn = || {
+            warn(
+                vm.ctx.exceptions.runtime_warning,
+                format!(
+                    "Ignoring unimportable $PYTHONBREAKPOINT: \"{}\"",
+                    envar.to_owned(),
+                ),
+                0,
+                vm,
+            )
+            .unwrap();
+            Ok(vm.ctx.none())
+        };
+
+        let (modulepath, attrname) = match envar.split('.').last() {
+            Some(last) => {
+                if last.eq(&envar) {
+                    ("builtins".to_owned(), envar.to_owned())
+                } else {
+                    (
+                        envar[..(envar.len() - last.len() - 1)].to_owned(),
+                        last.to_owned(),
+                    )
+                }
+            }
+            None => {
+                return print_unimportable_module_warn();
+            }
+        };
```

Comment:
let's reduce indent depth.

```suggestion
        let last = match envar.split('.').last().ok_or_else(print_unimportable_module_warn)?;
        let (modulepath, attrname) = if last.eq(&envar) {
            ("builtins".to_owned(), envar.to_owned())
        } else {
            (
                envar[..(envar.len() - last.len() - 1)].to_owned(),
                 last.to_owned(),
            )
        };
```

Please check the next block too.

Reply:
I think the error handling logic here is printing warning log and returning `vm.ctx.none`, so I think maybe error propagating can not be used here.

I'm trying to find good way to reduce indents, how about one?

```rust
        let last = match envar.split('.').last() {
            Some(last) => last,
            None => {
                return print_unimportable_module_warn();
            }
        };

        let (modulepath, attrname) = if last.eq(&envar) {
            ("builtins".to_owned(), envar.to_owned())
        } else {
            (
                envar[..(envar.len() - last.len() - 1)].to_owned(),
                last.to_owned(),
            )
        };

        let module = match vm.import(modulepath, None, 0) {
            Ok(module) => module,
            Err(_) => {
                return print_unimportable_module_warn();
            }
        };

        match vm.get_attribute_opt(module, attrname) {
            Ok(Some(hook)) => vm.invoke(hook.as_ref(), args),
            _ => print_unimportable_module_warn(),
        }
```

Reply:
Oh, I am sorry. you are right